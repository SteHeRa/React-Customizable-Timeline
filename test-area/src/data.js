const data = [
  {
    title: '2020',
    events: [
      {
        title: 'Wednesday, 16 September',
        subtitle: '九月十六日，星期三',
        content:
          '燕子去了，有再来的时候；杨柳枯了，有再青的时候；桃花谢了，有再开的时候。但是，聪明的，你告诉我，我们的日子为什么一去不复返呢？--是有人偷了他们罢：那是谁？又藏在何处呢？是他们自己逃走了罢：现在又到了哪里呢？',
        location: '北京',
        label: '匆匆 朱自清 散文',
        img: {
          url:
            'https://images.unsplash.com/photo-1600790078201-5490baf711d6?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60',
          alt: 'pic',
        },
      },
      {
        title: 'Tuesday, 16 June',
        subtitle: '六月十六日，星期二',
        content:
          '我不知道他们给了我多少日子；但我的手确乎是渐渐空虚了。在默默里算着，八千多日子已经从我手中溜去；像针尖上一滴水滴在大海里，我的日子滴在时间的流里，没有声音，也没有影子。我不禁头涔涔而泪潸潸了。',
        location: '上海',
        img: {
          url:
            'https://images.unsplash.com/photo-1587648030644-b53c19c433fb?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=800&q=60',
          alt: 'pic',
        },
      },
      {
        title: 'Friday, 27 March',
        subtitle: '副标题',
        content:
          '去的尽管去了，来的尽管来着；去来的中间，又怎样地匆匆呢？早上我起来的时候，小屋里射进两三方斜斜的太阳。太阳他有脚啊，轻轻悄悄地挪移了；我也茫茫然跟着旋转。于是--洗手的时候，日子从水盆里过去；吃饭的时候，日子从饭碗里过去；默默时，便从凝然的双眼前过去。我觉察他去的匆匆了，伸出手遮挽时，他又从遮挽着的手边过去，天黑时，我躺在床上，他便伶伶俐俐地从我身上跨过，从我脚边飞去了。等我睁开眼和太阳再见，这算又溜走了一日。我掩着面叹息。但是新来的日子的影儿又开始在叹息里闪过了。',
        location: '杭州',
        img: {
          url:
            'https://images.unsplash.com/photo-1600679621003-7d73ef359c8c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60.unsplash.com/photo-1600790078201-5490baf711d6?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60',
          alt: 'pic',
        },
      },
    ],
  },
  {
    title: '2019',
    events: [
      {
        title: 'Friday, 16 August',
        subtitle: '八月十六日，星期五',
        content:
          '在逃去如飞的日子里，在千门万户的世界里的我能做些什么呢？只有徘徊罢了，只有匆匆罢了；在八千多日的匆匆里，除徘徊外，又剩些什么呢？过去的日子如轻烟，被微风吹散了，如薄雾，被初阳蒸融了；我留着些什么痕迹呢？我何曾留着像游丝样的痕迹呢？我赤裸裸来到这世界，转眼间也将赤裸裸的回去罢？但不能平的，为什么偏要白白走这一遭啊？',
        location: '石家庄',
      },
      {
        title: 'Wednesday, 15 May',
        subtitle: '五月十五日，星期三',
        content: '你聪明的，告诉我，我们的日子为什么一去不复返呢',
        location: '成都',
      },
    ],
  },
  {
    title: '2018',
    events: [
      {
        title: 'Saturday, 29 December',
        subtitle: '九月二十九日，星期六',
        content:
          '这是在花园里。群花都还做她们的清梦。那微雨偷偷洗去她们的尘垢，她们的甜软的光泽便自焕发了。在那被洗去的浮艳下，我能看到她们在有日光时所深藏着的恬静的红，冷落的紫，和苦笑的白与绿。以前锦绣般在我眼前的，现有都带了黯淡的颜色。--是愁着芳春的销歇么？是感着芳春的困倦么？',
        location: '巴塞罗那',
      },
      {
        title: 'Friday, 5 October',
        subtitle: '十月五日，星期五',
        content:
          '大约也因那濛濛的雨，园里没了秾郁的香气。涓涓的东风只吹来一缕缕饿了似的花香；夹带着些潮湿的草丛的气息和泥土的滋味。园外田亩和沼泽里，又时时送过些新插的秧，少壮的麦，和成荫的柳树的清新的蒸气。这些虽非甜美，却能强烈地刺激我的鼻观，使我有愉快的倦怠之感。',
        location: '柏林',
      },
    ],
  },
];

export default data;
