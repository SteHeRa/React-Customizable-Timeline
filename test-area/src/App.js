import React from 'react';
import './App.css';
import data from './data'

import Timeline from './TimelineComponent/Timeline';

function App() {

  return (
    <div className="App">
      <Timeline
        data={data}
        primaryDarkColor="black"
        primaryLightColor="white"
        titleShape="circle"
        dotShape="diamond"
        lineStyle="dotted"
        primaryFont="monospace"
        animation={true}
      />
    </div>
  );
}

export default App;
