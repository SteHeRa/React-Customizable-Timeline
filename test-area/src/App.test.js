import React from 'react';
import { fireEvent, getByText, render, screen } from '@testing-library/react';
import App from './App';
import './__mocks__/intersectionObserverMock';

import data from './data';

describe('App', () => {
  it('renders correctly', () => {
    const app = render(<App />);

    expect(app).toMatchSnapshot();
  });

  it('has correct text in tiles', () => {
    render(<App />); //render the App component in test-area/src/App.js

    expect(screen.getByText('2020')).toBeInTheDocument(); //Check that the rendered component contains this text somewhere
    expect(screen.getByText('2019')).toBeInTheDocument();
    expect(screen.getByText('2018')).toBeInTheDocument();
    expect(screen.queryByText('2017')).not.toBeInTheDocument();
  });

  it('renders and derenders content on click', () => {
    render(<App />); //render the App component in test-area/src/App.js

    expect(screen.queryByText('16/09/2020 Label')).not.toBeInTheDocument(); //Check that the rendered component does not cotain this text
    expect(screen.queryByText('16/09/2020 Content')).not.toBeInTheDocument(); //Check that the rendered component does not cotain this text

    fireEvent.click(screen.getByText('16/09/2020 Subtitle')); //Simulate a click on the component (targeted using the event subtitle text)
    expect(screen.getByText('16/09/2020 Label')).toBeInTheDocument();
    expect(screen.getByText('16/09/2020 Content')).toBeInTheDocument(); //Check that following the click the rendered component contains this text

    fireEvent.click(screen.getByText('16/09/2020 Subtitle')); //Simulate another click on the component to derender the content
    expect(screen.queryByText('16/09/2020 Label')).not.toBeInTheDocument();
    expect(screen.queryByText('16/09/2020 Content')).not.toBeInTheDocument(); //Check that the rendered component does not cotain this text
  });

  it('renders the correct amount of items', () => {
    render(<App />);

    const timelineItems = screen.getAllByTestId('timelineItem');
    expect(timelineItems.length).toEqual(data.length);
  });
});

describe('ColorMatch', () => {
  it('should render the color in the color picker ', () => {});
});
