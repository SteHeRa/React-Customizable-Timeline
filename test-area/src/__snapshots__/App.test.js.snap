// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          style="transform: translate3d(-100%, 0, 0);"
        >
          <div
            class="sc-gtssRu lhASma"
          >
            <div
              class="sc-dlnjPT eFJCwd"
              data-testid="timelineItem"
            >
              <div
                style="transform: translate3d(0, -4px, 0) scaleY(1.02);"
              >
                <div
                  class="sc-bdnylx kkSrUA"
                >
                  <div>
                    2020
                  </div>
                </div>
              </div>
              <div
                class="css-q7lffx"
              >
                <div
                  class="sc-hKFyIo cxQuKu"
                >
                  <div
                    style="transform: scale3d(1.25, 0.75, 1);"
                  >
                    <div
                      style="cursor: pointer;"
                    >
                      <h4>
                        Wednesday, 16 September
                      </h4>
                      <h5>
                        16/09/2020 Subtitle
                      </h5>
                      <img
                        alt="pic"
                        src="https://images.unsplash.com/photo-1600790078201-5490baf711d6?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"
                        style="max-width: 400px; width: 100%; min-width: 100px; padding-bottom: 10px;"
                      />
                    </div>
                    <div
                      style="font-size: small; text-align: right;"
                    >
                      <span
                        aria-label="localtion"
                        role="img"
                      >
                        üìç
                      </span>
                       
                      16/09/2020 Location
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="css-q7lffx"
              >
                <div
                  class="sc-hKFyIo cxQuKu"
                >
                  <div
                    style="transform: scale3d(1.25, 0.75, 1);"
                  >
                    <div
                      style="cursor: pointer;"
                    >
                      <h4>
                        Tuesday, 16 June
                      </h4>
                      <h5>
                        16/06/2020 Subtitle
                      </h5>
                      <img
                        alt="pic"
                        src="https://images.unsplash.com/photo-1587648030644-b53c19c433fb?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=800&q=60"
                        style="max-width: 400px; width: 100%; min-width: 100px; padding-bottom: 10px;"
                      />
                    </div>
                    <div
                      style="font-size: small; text-align: right;"
                    >
                      <span
                        aria-label="localtion"
                        role="img"
                      >
                        üìç
                      </span>
                       
                      16/06/2020 Location
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="css-q7lffx"
              >
                <div
                  class="sc-hKFyIo cxQuKu"
                >
                  <div
                    style="transform: scale3d(1.25, 0.75, 1);"
                  >
                    <div
                      style="cursor: pointer;"
                    >
                      <h4>
                        Friday, 27 March
                      </h4>
                      <h5>
                        27/03/2020 Subtitle
                      </h5>
                      <img
                        alt="pic"
                        src="https://images.unsplash.com/photo-1600679621003-7d73ef359c8c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60.unsplash.com/photo-1600790078201-5490baf711d6?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"
                        style="max-width: 400px; width: 100%; min-width: 100px; padding-bottom: 10px;"
                      />
                    </div>
                    <div
                      style="font-size: small; text-align: right;"
                    >
                      <span
                        aria-label="localtion"
                        role="img"
                      >
                        üìç
                      </span>
                       
                      27/03/2020 Location
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="sc-dlnjPT eFJCwd"
              data-testid="timelineItem"
            >
              <div
                style="transform: translate3d(0, -4px, 0) scaleY(1.02);"
              >
                <div
                  class="sc-bdnylx kkSrUA"
                >
                  <div>
                    2019
                  </div>
                </div>
              </div>
              <div
                class="css-q7lffx"
              >
                <div
                  class="sc-hKFyIo cxQuKu"
                >
                  <div
                    style="transform: scale3d(1.25, 0.75, 1);"
                  >
                    <div
                      style="cursor: pointer;"
                    >
                      <h4>
                        Friday, 16 August
                      </h4>
                      <h5>
                        16/08/2019 Subtitle
                      </h5>
                    </div>
                    <div
                      style="font-size: small; text-align: right;"
                    >
                      <span
                        aria-label="localtion"
                        role="img"
                      >
                        üìç
                      </span>
                       
                      16/08/2019 Location
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="css-q7lffx"
              >
                <div
                  class="sc-hKFyIo cxQuKu"
                >
                  <div
                    style="transform: scale3d(1.25, 0.75, 1);"
                  >
                    <div
                      style="cursor: pointer;"
                    >
                      <h4>
                        Wednesday, 15 May
                      </h4>
                      <h5>
                        15/05/2019 Subtitle
                      </h5>
                    </div>
                    <div
                      style="font-size: small; text-align: right;"
                    >
                      <span
                        aria-label="localtion"
                        role="img"
                      >
                        üìç
                      </span>
                       
                      15/05/2019 Location
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="sc-dlnjPT eFJCwd"
              data-testid="timelineItem"
            >
              <div
                style="transform: translate3d(0, -4px, 0) scaleY(1.02);"
              >
                <div
                  class="sc-bdnylx kkSrUA"
                >
                  <div>
                    2018
                  </div>
                </div>
              </div>
              <div
                class="css-q7lffx"
              >
                <div
                  class="sc-hKFyIo cxQuKu"
                >
                  <div
                    style="transform: scale3d(1.25, 0.75, 1);"
                  >
                    <div
                      style="cursor: pointer;"
                    >
                      <h4>
                        Saturday, 29 December
                      </h4>
                      <h5>
                        29/12/2018 Subtitle
                      </h5>
                    </div>
                    <div
                      style="font-size: small; text-align: right;"
                    >
                      <span
                        aria-label="localtion"
                        role="img"
                      >
                        üìç
                      </span>
                       
                      29/12/2018 Location
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="css-q7lffx"
              >
                <div
                  class="sc-hKFyIo cxQuKu"
                >
                  <div
                    style="transform: scale3d(1.25, 0.75, 1);"
                  >
                    <div
                      style="cursor: pointer;"
                    >
                      <h4>
                        Friday, 5 October
                      </h4>
                      <h5>
                        05/10/2018 Subtitle
                      </h5>
                    </div>
                    <div
                      style="font-size: small; text-align: right;"
                    >
                      <span
                        aria-label="localtion"
                        role="img"
                      >
                        üìç
                      </span>
                       
                      05/10/2018 Location
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        style="transform: translate3d(-100%, 0, 0);"
      >
        <div
          class="sc-gtssRu lhASma"
        >
          <div
            class="sc-dlnjPT eFJCwd"
            data-testid="timelineItem"
          >
            <div
              style="transform: translate3d(0, -4px, 0) scaleY(1.02);"
            >
              <div
                class="sc-bdnylx kkSrUA"
              >
                <div>
                  2020
                </div>
              </div>
            </div>
            <div
              class="css-q7lffx"
            >
              <div
                class="sc-hKFyIo cxQuKu"
              >
                <div
                  style="transform: scale3d(1.25, 0.75, 1);"
                >
                  <div
                    style="cursor: pointer;"
                  >
                    <h4>
                      Wednesday, 16 September
                    </h4>
                    <h5>
                      16/09/2020 Subtitle
                    </h5>
                    <img
                      alt="pic"
                      src="https://images.unsplash.com/photo-1600790078201-5490baf711d6?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"
                      style="max-width: 400px; width: 100%; min-width: 100px; padding-bottom: 10px;"
                    />
                  </div>
                  <div
                    style="font-size: small; text-align: right;"
                  >
                    <span
                      aria-label="localtion"
                      role="img"
                    >
                      üìç
                    </span>
                     
                    16/09/2020 Location
                  </div>
                </div>
              </div>
            </div>
            <div
              class="css-q7lffx"
            >
              <div
                class="sc-hKFyIo cxQuKu"
              >
                <div
                  style="transform: scale3d(1.25, 0.75, 1);"
                >
                  <div
                    style="cursor: pointer;"
                  >
                    <h4>
                      Tuesday, 16 June
                    </h4>
                    <h5>
                      16/06/2020 Subtitle
                    </h5>
                    <img
                      alt="pic"
                      src="https://images.unsplash.com/photo-1587648030644-b53c19c433fb?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=800&q=60"
                      style="max-width: 400px; width: 100%; min-width: 100px; padding-bottom: 10px;"
                    />
                  </div>
                  <div
                    style="font-size: small; text-align: right;"
                  >
                    <span
                      aria-label="localtion"
                      role="img"
                    >
                      üìç
                    </span>
                     
                    16/06/2020 Location
                  </div>
                </div>
              </div>
            </div>
            <div
              class="css-q7lffx"
            >
              <div
                class="sc-hKFyIo cxQuKu"
              >
                <div
                  style="transform: scale3d(1.25, 0.75, 1);"
                >
                  <div
                    style="cursor: pointer;"
                  >
                    <h4>
                      Friday, 27 March
                    </h4>
                    <h5>
                      27/03/2020 Subtitle
                    </h5>
                    <img
                      alt="pic"
                      src="https://images.unsplash.com/photo-1600679621003-7d73ef359c8c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60.unsplash.com/photo-1600790078201-5490baf711d6?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"
                      style="max-width: 400px; width: 100%; min-width: 100px; padding-bottom: 10px;"
                    />
                  </div>
                  <div
                    style="font-size: small; text-align: right;"
                  >
                    <span
                      aria-label="localtion"
                      role="img"
                    >
                      üìç
                    </span>
                     
                    27/03/2020 Location
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="sc-dlnjPT eFJCwd"
            data-testid="timelineItem"
          >
            <div
              style="transform: translate3d(0, -4px, 0) scaleY(1.02);"
            >
              <div
                class="sc-bdnylx kkSrUA"
              >
                <div>
                  2019
                </div>
              </div>
            </div>
            <div
              class="css-q7lffx"
            >
              <div
                class="sc-hKFyIo cxQuKu"
              >
                <div
                  style="transform: scale3d(1.25, 0.75, 1);"
                >
                  <div
                    style="cursor: pointer;"
                  >
                    <h4>
                      Friday, 16 August
                    </h4>
                    <h5>
                      16/08/2019 Subtitle
                    </h5>
                  </div>
                  <div
                    style="font-size: small; text-align: right;"
                  >
                    <span
                      aria-label="localtion"
                      role="img"
                    >
                      üìç
                    </span>
                     
                    16/08/2019 Location
                  </div>
                </div>
              </div>
            </div>
            <div
              class="css-q7lffx"
            >
              <div
                class="sc-hKFyIo cxQuKu"
              >
                <div
                  style="transform: scale3d(1.25, 0.75, 1);"
                >
                  <div
                    style="cursor: pointer;"
                  >
                    <h4>
                      Wednesday, 15 May
                    </h4>
                    <h5>
                      15/05/2019 Subtitle
                    </h5>
                  </div>
                  <div
                    style="font-size: small; text-align: right;"
                  >
                    <span
                      aria-label="localtion"
                      role="img"
                    >
                      üìç
                    </span>
                     
                    15/05/2019 Location
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="sc-dlnjPT eFJCwd"
            data-testid="timelineItem"
          >
            <div
              style="transform: translate3d(0, -4px, 0) scaleY(1.02);"
            >
              <div
                class="sc-bdnylx kkSrUA"
              >
                <div>
                  2018
                </div>
              </div>
            </div>
            <div
              class="css-q7lffx"
            >
              <div
                class="sc-hKFyIo cxQuKu"
              >
                <div
                  style="transform: scale3d(1.25, 0.75, 1);"
                >
                  <div
                    style="cursor: pointer;"
                  >
                    <h4>
                      Saturday, 29 December
                    </h4>
                    <h5>
                      29/12/2018 Subtitle
                    </h5>
                  </div>
                  <div
                    style="font-size: small; text-align: right;"
                  >
                    <span
                      aria-label="localtion"
                      role="img"
                    >
                      üìç
                    </span>
                     
                    29/12/2018 Location
                  </div>
                </div>
              </div>
            </div>
            <div
              class="css-q7lffx"
            >
              <div
                class="sc-hKFyIo cxQuKu"
              >
                <div
                  style="transform: scale3d(1.25, 0.75, 1);"
                >
                  <div
                    style="cursor: pointer;"
                  >
                    <h4>
                      Friday, 5 October
                    </h4>
                    <h5>
                      05/10/2018 Subtitle
                    </h5>
                  </div>
                  <div
                    style="font-size: small; text-align: right;"
                  >
                    <span
                      aria-label="localtion"
                      role="img"
                    >
                      üìç
                    </span>
                     
                    05/10/2018 Location
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
